addons/hr_holidays/models/hr_leave.py

```diff
@@ -530,11 +530,11 @@ def _compute_has_mandatory_day(self):
             mandatory_days = self.employee_id._get_mandatory_days(
                 date_from.date(),
                 date_to.date())
-
             for leave in self:
                 domain = [
                     ('start_date', '<=', leave.date_to.date()),
                     ('end_date', '>=', leave.date_from.date()),
+                    ('department_ids', 'parent_of', leave.employee_id.department_id.id),
```

Comment:
I think this will not work if the employee department is False;
While checking locally, `self.env['hr.leave.mandatory.day'].search([('department_ids', 'parent_of', False)])` didn't return anything while some records with an empty field exist.
I guess you could do
```py
department_domain = [('department_ids', '=', False)]
if leave.employee_id.department_id:
    department_domain = expression.OR([
        department_domain,
        [('department_ids', 'parent_of', leave.employee_id.department_id.id)]
    ])
domain = expression.AND([domain, department_domain])